File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> m;
        for(int i = 0; i < strs.size(); i++) {
            string temp = strs[i];
            sort(temp.begin(), temp.end());
            
            if(!m[temp].size()){
                m[temp] = {strs[i]};
            } else {
                m[temp].push_back(strs[i]);
            }
        }
        vector<vector<string>> res;
        
        for(auto item : m) {
            res.push_back(item.second);
        }
        
        return res;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anag = defaultdict(list)

        for i in strs:
            temp = sorted(i)
            anag[''.join(temp)].append(i)
        res = []
        for i in anag:
            res.append(anag[i])
        return res